hw/drivers/crypto/crypto_nrf52/src/crypto_nrf52.c

```diff
@@ -34,27 +34,34 @@ struct aes_128_data {
 static bool
 nrf52_crypto_ecb_start(const struct aes_128_data *crypto_data)
 {
-    int ctr = 0x100000;
-    bool rc = true;
+    bool rc;
+    uint32_t end;
+    uint32_t err;
 
+    /* Just to be sure, always stop ECB and clear END/ERROR registers */
+    NRF_ECB->TASKS_STOPECB = 1;
+    NRF_ECB->EVENTS_ENDECB = 0;
+    NRF_ECB->EVENTS_ERRORECB = 0;
+
+    /* Set the data pointer and start ECB */
     NRF_ECB->ECBDATAPTR = (uint32_t)crypto_data;
     NRF_ECB->TASKS_STARTECB = 1;
 
-    while (ctr-- > 0) {
-        if (NRF_ECB->EVENTS_ENDECB) {
+    while (1) {
+        end = NRF_ECB->EVENTS_ENDECB;
+        err = NRF_ECB->EVENTS_ERRORECB;
+        if (end || err) {
+            if (err) {
```

Comment:
You could eliminate this `if` / `else` and `rc` variable by doing `return (err == 0);` at the end, although it would be less readable I suppose. I agree with @andrzej-kaczmarek on using `bool` for `in_use`, otherwise everything looks ok.